### ISSUE TITLE
2.4.11 Unusual scenario: Very short links with a thick underline focus state

### ISSUE THREAD

**Issue by 36degrees** [w3c/wcag#1329]
Raising this in response to:

> The working group is interested in feedback about the minimum area metric, and if there are unusual scenarios where visible indicators are caught by the wording.

We've been looking at how the new focus indication criteria would apply to links in the GOV.UK Design System.

![A link with the text 'Hello world'](https://user-images.githubusercontent.com/121939/90785766-a4095d80-e2fa-11ea-9f14-e6b5f3cc5d5a.png)

Based on the dimensions in Chrome's developer tools, the bounding box for this link is 94px wide and 25px high.

Based on the current example calculations (and ignoring #1324) this means we'd be looking for a minimum focus indication area of 94 + 94 + 25 +25 = 238px.

Our current link focus state uses a 4px black border along the bottom edge:

![A link with the text 'Hello world', focused. The focused link has a yellow background and a 4px black border along the bottom edge.](https://user-images.githubusercontent.com/121939/90785811-b2f01000-e2fa-11ea-8f0d-a6620ac8d3d9.png)

(The yellow is not considered part of the focus indication area for links on light backgrounds, as it does not meet the 3:1 change of contrast requirement)

94px × 4px = 376px, which is greater than 238px, so we're fine.

However, when links are narrower than they are tall the maths no longer works:

![A link with the text 'aa'](https://user-images.githubusercontent.com/121939/90785769-a53a8a80-e2fa-11ea-9454-3ff093eb33bf.png)

This link, with the text 'aa', is still 25px high but now only 19px wide. 19 + 19 + 25 + 25 = a 88px minimum focus indication area.

![A link with the text 'aa', focused. The focused link has a yellow background and a 4px black border along the bottom edge.](https://user-images.githubusercontent.com/121939/90785770-a66bb780-e2fa-11ea-822f-a1a0d02723ca.png)

19px × 4px = only 76px, less than the calculated 88px minimum required.

I would potentially consider this an unusual scenario where a visible indicator is caught by the wording.

This is loosely related to #1328, as although we may be able to rely on links generally being longer than they are wide (on the assumption that the link text should be descriptive) if a link is broken over multiple lines you can end up with an orphaned short word:

<img width="642" alt="Screenshot 2020-08-14 at 15 24 50" src="https://user-images.githubusercontent.com/121939/90787340-8b01ac00-e2fc-11ea-846d-e5b28ce41308.png">

**Comment by mraccess77** [comment]
My personal opinion is that when a link is split over multiple lines that it should be treated as one unit and not multiple units for calculation of area.

**Comment by alastc** [comment]
I agree with Jon about #1328, but in the case of an independant small link, I'm not sure what to do that would not make the size-measure much more complex. Any ideas?

**Comment by alastc** [comment]
Hi everyone,

Based on an update which was surveyed/discussed in the last two weeks, we've updated the size metric to read:

<hr>

**Minimum area**: The area of the focus indicator is either:

- at least as large as the area of a 1 CSS pixel thick perimeter of the unfocused control;
- at least as large as a 4 CSS pixels border along the shortest side, and no thinner than 2 CSS pixels.

<hr>

Survey: https://www.w3.org/2002/09/wbs/35422/wcag22-focus-appearance-updates2/results#xq1
Minutes: https://www.w3.org/2021/01/19-ag-minutes.html#t02 
Understanding document: https://w3c.github.io/wcag/understanding/focus-appearance-minimum.html (Should update soon)

@36degrees I think that either the "i" link will now pass, or it is very close to doing so. 

Based on that we'll close this issue, but please raise a new one if there are problems with the new text.
